hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/protocol/ClientProtocol.java

```diff
@@ -968,5 +969,14 @@ Map<DatanodeDetails, OzoneInputStream>> getKeysEveryReplicas(
    */
   String createSnapshot(String volumeName,
       String bucketName, String snapshotName) throws IOException;
-  
+
+  /**
+   * List snapshots in a volume/bucket.
+   * @param volumeName volume name
+   * @param bucketName bucket name
+   * @return list of snapshot name
+   * @throws IOException
+   */
+  List<OzoneSnapshot> listSnapshot(String volumeName, String bucketName)
```

Comment:
Just curious why this API is not paginated?

Reply:
Ok. But, I think this patch will be too large. I'll create another jira ticket to make this API paginated.